# Notebook for Data visualization, Data_preprocessing, 

# Importing Necessary Model from Project Structure

In [15]:
import os

In [86]:
from Scripts import read_data as rd


In [87]:
#Define path for raw data 

In [88]:
base = os.path.dirname(os.path.abspath('__file__'))
path_nepsedata = os.path.join(base, "data/raw/Agricultural_development_bank_limited.xlsx")
path_nepsenews = os.path.join(base, "data/raw/raw_news_data.csv")
path_score = os.path.join(base, "data/raw/news_score.csv")

# Read data from files

In [92]:
df_stock = rd.read_dataset(path_nepsedata, file_type='xlsx')

In [93]:
df_stock.head()

Open  Close  Volume  Turnover
Date                                     
2011-03-20   132    133    1507    198979
2011-03-24   135    133    3116    415035
2011-03-25   133    135    2560    342820
2011-03-27   135    134    2110    284040
2011-03-28   134    130    1288    169336

In [94]:
df_stock.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 2222 entries, 2011-03-20 to 2021-01-13
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Open      2222 non-null   int64
 1   Close     2222 non-null   int64
 2   Volume    2222 non-null   int64
 3   Turnover  2222 non-null   int64
dtypes: int64(4)
memory usage: 86.8 KB


1) Dataframe contain 2222 days values of stock from 2011-03-20 to 2021-01-13
2) The main features are taken out from the dataset

In [56]:
df_news = rd.read_dataset(path_nepsenews)

In [57]:
df_news.head()

Title
Date                                                         
2019-11-15  Nepal Insurance Company posts net profit of Rs...
2019-11-15  Look at the major highlights from the first qu...
2019-11-15  Sanima Life Insurance and Mahalaxmi Life Insur...
2019-11-15  NMB Bank was traded the most this week; Sabaik...
2019-11-15  Ganapati Microfinance announces 3rd AGM on Pou...

In [60]:
df_scores =rd.read_dataset(path_score)

In [63]:
df_scores.head()

Title  \
Date                                                            
2019-11-15  Nepal Insurance Company posts net profit of Rs...   
2019-11-15  Look at the major highlights from the first qu...   
2019-11-15  Sanima Life Insurance and Mahalaxmi Life Insur...   
2019-11-15  NMB Bank was traded the most this week; Sabaik...   
2019-11-15  Ganapati Microfinance announces 3rd AGM on Pou...   

                                                       scores  compound  
Date                                                                     
2019-11-15  {'neg': 0.0, 'neu': 0.861, 'pos': 0.139, 'comp...    0.4404  
2019-11-15  {'neg': 0.083, 'neu': 0.917, 'pos': 0.0, 'comp...   -0.2263  
2019-11-15  {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound...    0.0000  
2019-11-15  {'neg': 0.0, 'neu': 0.893, 'pos': 0.107, 'comp...    0.3400  
2019-11-15  {'neg': 0.0, 'neu': 0.667, 'pos': 0.333, 'comp...    0.7906

In [72]:
df_scores = df_scores.drop(['Title','scores'], axis = 1)

# Data Cleanning, joining and preprocessing 

In [81]:
final_data = rd.preprocess_data(rd.join_dataframe(dataframe, df_scores))

In [82]:
final_data.head()

Open  Close  Volume  Turnover  compound
2011-03-20  132.0  133.0  1507.0  198979.0       NaN
2011-03-24  135.0  133.0  3116.0  415035.0       NaN
2011-03-25  133.0  135.0  2560.0  342820.0  0.135946
2011-03-27  135.0  134.0  2110.0  284040.0  0.098667
2011-03-28  134.0  130.0  1288.0  169336.0  0.142644

In [85]:
final_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2210 entries, 2011-03-20 to 2021-01-13
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Open      2210 non-null   float64
 1   Close     2210 non-null   float64
 2   Volume    2210 non-null   float64
 3   Turnover  2210 non-null   float64
 4   compound  2210 non-null   float64
dtypes: float64(5)
memory usage: 103.6+ KB


In [84]:
final_data['compound'] = final_data['compound'].fillna(0)

# Storing Final dataframe into new CSV file¶

In [96]:
final_data.to_csv('final_dataset.csv',index=True)